components/org.wso2.carbon.identity.api.server.authenticators/org.wso2.carbon.identity.api.server.authenticators.v1/src/main/java/org/wso2/carbon/identity/api/server/authenticators/v1/impl/LocalAuthenticatorConfigBuilderFactory.java

```diff
@@ -0,0 +1,144 @@
+            throws AuthenticatorMgtClientException {
+
+        String authenticationType = AuthenticatorPropertyConstants.AuthenticationType.IDENTIFICATION.toString();
+        if (config.getAuthenticationType() != null) {
+            authenticationType = config.getAuthenticationType().toString();
+        }
+        UserDefinedLocalAuthenticatorConfig authConfig = new UserDefinedLocalAuthenticatorConfig(
+                AuthenticatorPropertyConstants.AuthenticationType.valueOf(authenticationType));
+        authConfig.setName(config.getName());
+        authConfig.setDisplayName(config.getDisplayName());
+        authConfig.setEnabled(config.getIsEnabled());
+        authConfig.setEndpointConfig(buildEndpointConfig(config.getEndpoint()));
+
+        return authConfig;
+    }
+
+    /**
+     * Builds the UserDefinedLocalAuthenticatorConfig from the user defined local authenticator update request.
+     *
+     * @param config The user defined local authenticator update request.
+     * @param existingConfig The existing user defined local authenticator configuration.
+     * @return The user defined local authenticator configuration.
+     * @throws AuthenticatorMgtClientException If an error occurs while building the configuration.
+     */
+    public static UserDefinedLocalAuthenticatorConfig build(UserDefinedLocalAuthenticatorUpdate config,
+                LocalAuthenticatorConfig existingConfig) throws AuthenticatorMgtClientException {
+
+        UserDefinedLocalAuthenticatorConfig authConfig = new UserDefinedLocalAuthenticatorConfig(
+                resolveAuthenticationType(existingConfig));
+        authConfig.setName(existingConfig.getName());
+        authConfig.setDisplayName(config.getDisplayName());
+        authConfig.setEnabled(config.getIsEnabled());
+        authConfig.setEndpointConfig(buildEndpointConfig(config.getEndpoint()));
+
+        return authConfig;
+    }
+
+    private static UserDefinedAuthenticatorEndpointConfig buildEndpointConfig(Endpoint endpointConfig)
+            throws AuthenticatorMgtClientException {
+
+        try {
+            UserDefinedAuthenticatorEndpointConfig.UserDefinedAuthenticatorEndpointConfigBuilder endpointConfigBuilder =
+                    new UserDefinedAuthenticatorEndpointConfig.UserDefinedAuthenticatorEndpointConfigBuilder();
+            endpointConfigBuilder.uri(endpointConfig.getUri());
+            endpointConfigBuilder.authenticationType(endpointConfig.getAuthentication().getType().toString());
+            endpointConfigBuilder.authenticationProperties(endpointConfig.getAuthentication().getProperties()
+                    .entrySet().stream().collect(Collectors.toMap(
+                            Map.Entry::getKey, entry -> entry.getValue().toString())));
+            return endpointConfigBuilder.build();
+        } catch (NoSuchElementException | IllegalArgumentException e) {
```

Comment:
Modify endpoint builder to check those exceptions and throw client exception

Reply:
The endpointConfigBuilder is common model for both federated and local authenticators, hence will not throw the exception from that model class.